vm/interpreter/src/vm.rs

```diff
@@ -98,14 +87,14 @@ where
     P: NetworkParams,
 {
     #[allow(clippy::too_many_arguments)]
-    pub fn new<R, C>(
+    pub fn new<R>(
         root: Cid,
         store_arc: DB,
         epoch: ChainEpoch,
         rand: &R,
         base_fee: BigInt,
         network_version: NetworkVersion,
-        circ_supply_calc: C,
+        circ_supply_calc: impl FnOnce(ChainEpoch, &StateTree<&DB>) -> Result<TokenAmount, anyhow::Error>,
```

Comment:
Let's not use the shorthand with such a long type. It'd be nicer to put it in the `where` section.